code/upaya/README.md

```diff
@@ -1,34 +1,79 @@
 upaya
-================
-The recognition of Identity is an often misunderstood concept in our Democracy. 
+=====
+
+The recognition of Identity is an often misunderstood concept in our Democracy.
 
 
 ### Dependencies
 
-- Ruby 2.0+ (try [rbenv](https://github.com/rbenv/rbenv) or [rvm](https://rvm.io/))
+- Ruby 2.3.0 (try [rbenv](https://github.com/rbenv/rbenv) or
+  [rvm](https://rvm.io/))
 - Bundler (`gem install bundler`)
+- postgresql (`brew install postgresql` or `apt-get install postgresql`)
+
+### Mac
+
 - [Vagrant](https://www.vagrantup.com/downloads.html)
 - [Virtualbox](https://www.virtualbox.org/wiki/Downloads) or [VMware Fusion](https://www.vmware.com/products/fusion)
 
-Or, try the [Laptop](https://github.com/18F/laptop) script for setting up a slick development environment.
+Or, try the [Laptop](https://github.com/18F/laptop) script for setting up a
+slick development environment.
 
 ### Getting Started
-##### :construction: VM environment coming soon! :construction: 
+
+Run the following command to set up the environment:
+
+    $ bin/setup
+
+This basically runs bundler to install the required gems and sets up the
+database.  If that is successful you can run the app server with:
+
+    $ bin/rails server
+
+### Running Tests
+
+To run all the tests:
+
+    $ bundle exec rspec --require rails_helper
+
+See RSpec [docs](https://relishapp.com/rspec/rspec-core/docs/command-line) for
+more information.
+
+### Docker
+
+ * Install the [Docker Toolbox](https://www.docker.com/products/docker-toolbox)
+ * Ensure your docker VM is running: `docker-machine start default`
+ * Fire up postgres: `docker run --name postgres -e POSTGRES_PASSWORD=$(openssl rand -hex 16) -d postgres`
+ * Build your development container: `docker build -t upaya .`
+ * Initialize your database: `docker run --link postgres --rm upaya rake db:setup`
+ * Run your app: `docker run --link postgres --rm upaya`
+ * Run your app, mounting your local repo in the container: `docker run --link postgres --rm -v $(pwd):/upaya`
+ * Run a shell in that container instead: `docker run --link postgres --rm -ti -v $(pwd):/upaya bash`
```

Comment:
Please do! I was just porting this PR over when I realized it wasn't built with docker compose.